### PR TITLE
feature/cf worker refresh nightly

### LINKED ISSUE #2: CF Worker to refresh KV store content on a nightly basis
- https://developers.cloudflare.com/workers/runtime-apis/scheduled-event/
- Cron expression TEST `0 23 * * *`, PROD `0 5 * * *` UTC time
- API path at `app/routes/api.refresh-content.tsx`
- Worker should be named `refresh-trigger` under the `workers` dir
- Promote worker to TEST and PROD envs 
- By default, do not supply purge param, but allow config through env

### PR DESCRIPTION
**Pull request by michelle-lai-swl**
Uses Github Actions to create a worker that refreshes the key-value store every day for the test and prod environments.

- 5:00 AM refresh for prod
- 11:00 PM refresh for test
- Purge is set as None by default
- Timeout is currently set to 120s (currently takes about 50-60s)

Used this solution for solving the separation by schedule: https://github.com/orgs/community/discussions/25928

Let me know @richardhunghhw if we can use what I have now, or if I should use (op://${{ vars.VAULT_NAME }}/HT Basic/hostname) as we'd only be able to use it for test atm?

### CODE REVIEW

File: .github/workflows/refresh-trigger.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    steps:
+      - name: Set host based on env
+        run: |
+            if [ "${{ github.event.schedule }}" = "0 5 * * *" ]; then
+              echo "HOST=https://www.hungrytapirldn.com" >> $GITHUB_ENV
+            elif [ "${{ github.event.schedule }}" = "0 23 * * *" ]; then
+              echo "HOST=https://develop.hungrytapir-store.pages.dev" >> $GITHUB_ENV
+            fi
+      - name: Set purge variable
+        run: echo "PURGE=" >> $GITHUB_ENV # set as 'True' for purge
```
Comment (richardhunghhw) [inline .github/workflows/refresh-trigger.yml]: Please use environment var `REFRESH_PURGE`

File: .github/workflows/refresh-trigger.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+              echo "URL=${{ env.HOST }}/api/refresh-content?purge=true" >> $GITHUB_ENV
+            else
+              echo "URL=${{ env.HOST }}/api/refresh-content" >> $GITHUB_ENV
+            fi
+      - name: Load secrets from 1Password
+        uses: 1Password/load-secrets-action@v1
+        env:
+            OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN_TEST }}
+            BASIC_AUTH_USERNAME: 'op://${{ vars.VAULT_NAME }}/HT Basic/username'
+            BASIC_AUTH_PASSWORD: 'op://${{ vars.VAULT_NAME }}/HT Basic/credential'
```
Comment (richardhunghhw) [inline .github/workflows/refresh-trigger.yml]: Created different vault variables `VAULT_NAME_TEST` and `VAULT_NAME_PROD`. Remember to get the url from `op://${{ vars.VAULT_NAME }}/HT Basic/hostname`

Comment (michelle-lai-swl) [reply .github/workflows/refresh-trigger.yml]: Fixed all comments, works for test but not working for prod: https://github.com/hungrytapirldn/hungrytapirldn.com/actions/runs/5581045902/jobs/10198763332
@richardhunghhw let me know if I got the OP_SERVICE_ACCOUNT_TOKEN wrong

File: .github/workflows/refresh-trigger.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    steps:
+      # - name: Set env based on schedule time
+      #   run: |
+      #       if [ "${{ github.event.schedule }}" = "0 5 * * *" ]; then
+      #         echo "ENVIRONMENT=PROD" >> $GITHUB_ENV
+      #       elif [ "${{ github.event.schedule }}" = "0 23 * * *" ]; then
+      #         echo "ENVIRONMENT=TEST" >> $GITHUB_ENV
+      #       fi
+      - name: Set env based on schedule time
+        run: echo "ENVIRONMENT=PROD" >> $GITHUB_ENV
```
Comment (richardhunghhw) [inline .github/workflows/refresh-trigger.yml:21]: Pls fix 😄

Comment (richardhunghhw) [inline .github/workflows/refresh-trigger.yml]: pls fix 😄

File: .github/workflows/refresh-trigger.yml (hunk 2 above)
Comment (richardhunghhw) [reply .github/workflows/refresh-trigger.yml]: Good point, use OP_SERVICE_ACCOUNT_TOKEN_PROD for the production one

Comment (michelle-lai-swl) [reply .github/workflows/refresh-trigger.yml]: https://github.com/hungrytapirldn/hungrytapirldn.com/actions/runs/5604647339/jobs/10252871521 Prod works

**Comment by michelle-lai-swl** [comment]
Please squash and merge after final review @richardhunghhw

**Pull request merged**